plugins/rbac/src/hooks/usePermissionPolicies.ts

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { useAsync, useAsyncRetry } from 'react-use';
+
+import { useApi } from '@backstage/core-plugin-api';
+
+import { rbacApiRef } from '../api/RBACBackendClient';
+import { getPermissionsData } from '../utils/rbac-utils';
+
+export const usePermissionPolicies = (entityReference: string) => {
+  const rbacApi = useApi(rbacApiRef);
+  const {
+    loading: policiesLoading,
+    value: policies,
+    retry,
+  } = useAsyncRetry(async () => {
+    return await rbacApi.getPolicies();
+  });
+  const { value: permissionPolicies } = useAsync(async () => {
```

Comment:
would be good to check on loading here and in general good to check on error cased as well

Reply:
How do we want to show the user if there is failure in API call ?
Note: Each card makes different API calls cc @parvathyvr @ShiranHi

Reply:
> How do we want to show the user if there is failure in API call ? Note: Each card makes different API calls cc @parvathyvr @ShiranHi

@debsmita1 Does it make sense to use the [error banner](https://www.figma.com/file/wlH5SQTQiIMHp5T0t7P4aU/Backstage-Design-System-(Community)?type=design&node-id=137-772&mode=design&t=is0zKoWImNDw28Ap-4) in backstage design system for this scenario?

Reply:
![Screenshot 2023-11-29 at 10 29 49 PM](https://github.com/janus-idp/backstage-plugins/assets/22490998/cf04bf10-8df5-454f-8c3a-50caa580ec13)

Reply:
@debsmita1 I know this is an edge case and won't happen ideally but should we show the error above the table under each card rather than inside of the table?

@parvathyvr WDYT?

Reply:
@invincibleJai i thought we could replace the 'No records to display' with the error message as @debsmita1 has implemented-But I am okay with the error being above the empty table as well- If that is a better option!
CC @ShiranHi do we have a preference/guideline about this?

Reply:
okay nice if you have considered this 👍

Reply:
Updated

![Screenshot 2023-11-30 at 1 52 50 PM](https://github.com/janus-idp/backstage-plugins/assets/22490998/1031a5ab-4389-4860-a577-90476c355819)

Reply:
@debsmita1 it looks good! Can we add a bit of a 16px space between the error message and the table ?
Also, it'll be great if a word will not break into two lines "grou-ps" and "poli-cies" like it is right now. 
Thanks for all the changes!

Reply:
![Screenshot 2023-12-01 at 12 26 14 AM](https://github.com/janus-idp/backstage-plugins/assets/22490998/a043a741-542a-4891-9530-1633c97b5ea1)

Reply:
A small note- @debsmita1 we are showing 'No members found' and 'No permissions found' on the empty table.
But ideally since the table heading are 'Users and groups' and 'Permission policies' i think it should be 'No users and groups found' and 'No permission policies found'.
But i also wonder if it's easier to show the generic message' No records found'.

Reply:
> The word breaks on small screens, what do you suggest here ?

I would suggest changing to 1 column per table to avoid text breaks as well as table misalignment.
Another reason to avoid 2 column composition is to reduce the cognitive load and showing two elements next to each other is usually for comparing them (which is not the case here).
So,
TL;DR
Moving to 1 column layout will help with:

- Get rid of the text breaks
- Allow easier data consumption with no distractions